kolibri/plugins/coach/assets/src/routes/index.js

```diff
@@ -11,32 +11,36 @@ import { ClassesPageNames } from '../../../../learn/assets/src/constants';
 import { PageNames } from '../constants';
 import reportRoutes from './reportRoutes';
 import planRoutes from './planRoutes';
+import { classIdParamRequiredGuard } from './utils';
 
 export default [
   ...planRoutes,
   ...reportRoutes,
   {
-    path: '/facilities',
+    name: 'AllFacilitiesPage',
+    path: '/:subtopicName?/facilities',
     component: AllFacilitiesPage,
+    props: true,
     handler() {
       store.dispatch('notLoading');
     },
   },
   {
-    path: '/classes',
+    path: '/:facility_id?/classes',
```

Comment:
You are missing a `subtopicName` parameter in the URL.

This means that while your changes do kind of work in the multi-class case, if I refresh the URL on this class select page, then the state is lost, and I get redirected back to Home after selecting the class.

Reply:
great point, thank you!